pytorch_lightning/trainer/trainer.py

```diff
@@ -2287,6 +2289,47 @@ def terminate_on_nan(self, val: bool) -> None:
         )
         self._terminate_on_nan = val  # : 212
 
+    def _call_callbacks_on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict[str, dict]:
```

Comment:
@jjenniferdai - this is where having a separate `state_dict/load_state_dict` to collect and load states from individual components vs `on_save/load_checkpoint` will be useful. one operates on the local state (per component), one operates on the global checkpoint (across all components)

Reply:
planning to file this issue this week!